### PR TITLE
use temp dir for jinja2 cache

### PR DESCRIPTION
**Pull request by fanjinfei**
Fixes #

### Features:

- [ ] includes tests covering changes
- [ ] includes updated documentation
- [ ] includes user-visible changes
- [ ] includes API changes
- [ ] includes bugfix for possible backport

Please [X] all the boxes above that apply

### CODE REVIEW

**Comment by wardi** [comment]
will this spam empty directories in tmp when we run the tests? (when the config is reloaded multiple times)

**Comment by TkTech** [comment]
@wardi set `jinja2_cache_dir` for tests

**Comment by wardi** [comment]
yes, adding a fixed directory to test-core.ini would be one fix, cleaning up when config is reloaded would be another

File: ckan/config/environment.py [hunk 1]
```diff
@@ -373,9 +373,19 @@ def template_loaded(template):
     # Create Jinja2 environment
     cache_dir = config.get('jinja2_cache_dir', None)
     if not cache_dir:
-        cache_dir = tempfile.mkdtemp()
+        global TEMP_JINJA2_CACHE_DIR
```
Comment (TkTech) [inline ckan/config/environment.py]: Nooooooo, don't do this, go back to the last version you had which was perfect. You do NOT need or want to clean the cache directory if one is specified, Jinja is doing that for you with bytecode verification.

Globals are *BAD* and should be avoided at all costs.

**Comment by fanjinfei** [comment]
@wardi shall I just add a fixed directory for test-core.ini?

**Comment by wardi** [comment]
This looks fine, is there a corresponding PR on ckan/ckan with a description of the issue this solves?

**Pull request merged**